runtime/scripts/part.js

```diff
@@ -1330,6 +1344,7 @@ if(res) { \
                     }
                 };
             }
+            this.events.trigger('markAlternatives', alt, exec_path);
```

Comment:
`this` is not what you think it is here. `part` is a reference to the part object.

If we do need an event here, it should be called `mark_alternative`. I think it'd be most useful at the top of the function.